wemake_python_styleguide/violations/oop.py

```diff
@@ -671,3 +673,33 @@ def get_attribute(self):
 
     error_template = 'Found unpythonic getter or setter'
     code = 615
+
+
+@final
+class BuggySuperContextViolation(ASTViolation):
+    """
+    Calling super() in buggy context.
+
+    Reasoning:
+        Call to `super()` without arguments will cause unexpected `TypeError`
+        in a number of specific contexts, e.g. dict/set/list comprehensions
+        and generator expressions.
```

Comment:
and nested functions.

Reply:
I will not amend that, as _e.g._ here indicates perfectly well that the list is incomplete.